[PIM-44] 사이드바 컴포넌트 구현

SeonDahye opened the pull request:
## 사이드바 컴포넌트 구현 <!-- 소셜 로그인 구현 -->

### 지라 티켓 번호
PIM-44

### PR 타입(하나 이상의 PR 타입을 선택해주세요)
- [x] 기능 추가
- [ ] 기능 삭제
- [ ] 버그 수정
- [ ] 의존성, 환경 변수, 빌드 관련 코드 업데이트

### 구현 내용
<!-- 구글 소셜 로그인 연동 -->
<!-- 테스트 결과 포함 -->

**/workspace-detail**
- 사이드바 컴포넌트로 구현
![image](https://user-images.githubusercontent.com/81613151/212584380-b325ac23-ec95-45c1-bb23-af5b2b109550.png)
- 사이드바 스토리북
![image](https://user-images.githubusercontent.com/81613151/212584491-c4c3e8af-ff56-4d30-932f-f0b05e803e7e.png)
- 사이드바 안의 컴포넌트 스토리북(이미지, 텍스트)
![image](https://user-images.githubusercontent.com/81613151/212584550-71a4d4cd-f3af-4988-8259-2a014d283624.png)
![image](https://user-images.githubusercontent.com/81613151/212584568-2de0da08-6eae-4f02-96c6-7988a11088b6.png)

### 리뷰 포인트
스토리북이 잘 작동하는지 확인해주세요!

### TODO
테스트 코드 작성을 할 예정입니다. 그리고 워크스페이스 페이지의 나머지 컴포넌트들도 스토리북을 작성할 예정입니다.

CHOSIYEON reviewed (approved):
화면이 참 매끄럽고 예뻐요!
변수명을 잘 지으신거같고, 주석이 있어서 코드 이해가 쉬워용 

다만 제 화면에서 봤을 때, 워크스페이스 보드 아이템끼리 겹치는 문제가 발생했어요 ~! 같이 고쳐보아욧

SeonDahye commented:
> 다만 제 화면에서 봤을 때, 워크스페이스 보드 아이템끼리 겹치는 문제가 발생했어요 ~! 같이 고쳐보아욧

리뷰 감사합니당 👍❣
아이템 겹치는 문제 해결했습니다 ! 💞

CHOSIYEON reviewed (approved):
예뻐요 예뻐요 수정해주셔서 감사해욧 !

Seyiul reviewed (approved):
기존에 구상한대로 UI가 잘 구현된 것 같습니다. 수고하셨습니다 👏👏👏👏👏

SeonDahye committed:
Merge remote-tracking branch 'origin/develop' into feat/PIM-44

SeonDahye committed:
Merge branch 'develop' into feat/PIM-44

SeonDahye committed:
feat: 사이드 바 컴포넌트 스토리북 생성

- TODO: props 문제 해결

SeonDahye committed:
style: props 이슈 #1

SeonDahye committed:
fix: props 이슈 #1

- member array props 에러 해결
- TODO: arrary 중첩 문제

SeonDahye committed:
fix: arrary 중첩 문제 #1

- members 이중으로 array 만들어지는 문제 해결

SeonDahye committed:
feat: 사이드바 관련 컴포넌트 생성 및 스토리북 적용

- SubTitle, WorkspaceImg 컴포넌트 생성 및 스토리북 적용

Seyiul committed:
feat: 로그인 api 구현

Seyiul committed:
feat: form validation 추가

SeonDahye committed:
style: develop에서 머지

Seyiul committed:
feat: 로그인 validation 추가

- 이메일을 이메일 형식으로 입력하지않은 경우
- 비밀번호를 영문/숫자/특수문자의 조합으로 입력하지 않은 경우

로그인이 제한될 수 있도록 구현

SeonDahye committed:
style: workspaceimg 컴포넌트 props 변경

- radius props 선택으로 변경(둥근, 원, 사각형)

Seyiul committed:
feat: utils > validation 추가

- utils 파일 생성 후, 로그인 및 회원가입에 필요한 validation을 저장

CHOSIYEON commented on src/pages/workspace/default/index.tsx:
```diff
@@ -0,0 +1,56 @@
+  ItemBoardName,
+  ProfileImages,
+  ProfileImage,
+} from './styles';
```
import * as S 로 해서 한번에 받아오면 더 코드가 간결해질 것 같아요!!

SeonDahye replied:
네 수정완료했습니당 ! ✨✨

CHOSIYEON commented on src/pages/workspace/detail/index.tsx:
```diff
@@ -0,0 +1,129 @@
+                  <Title margin={'30px 20px'}>{board.title}</Title>
+                </Item>
+              </Grid>
+            ))}
```
Warning: Each child in a list should have a unique "key" prop.
실행했을 때 다음과 같은 경고가 뜹니다 ! map을 사용할땐, key값 지정을 해주면 더 좋을 것 같아요

SeonDahye replied:
네 수정완료했습니당 ! ✨✨

CHOSIYEON commented on src/pages/workspace/default/styles.ts:
```diff
@@ -0,0 +1,87 @@
+  font-family: 'LINESeedKR-Bd';
+  color: #4f4e4e;
+`;
+const CreateButton = styled.button`
+  font-family: 'LINESeedKR-Rg';
+  background-color: #fca4be;
+  color: white;
+  padding: 0.6rem 2.5rem;
+  border-radius: 0.5rem;
+  font-size: 15px;
+  font-weight: 500;
+  box-shadow: 0 2px 3px rgba(0, 0, 0, 0.3);
+`;
```
스토리북 버튼 재사용하면 좋을 것 같아요!

SeonDahye replied:
버튼이 여러가지 종류여서 재사용이 힘드네요 ... 😂 어떻게 할까요 ㅎㅎㅎㅎㅎㅎㅎㅎ

CHOSIYEON reviewed (approved):
구현을 아주 잘하셨네요!!! 화면이동도 잘 됩니다!!! ❤️❤️❤️

SeonDahye committed:
style: map에 key 추가, 컨벤션 피드백 수정

Seyiul committed:
feat: 회원가입 UI 구현

Seyiul committed:
Revert "feat: 회원가입 UI 구현"

This reverts commit 849cdb40d4702b3e54b11865d38c5018d0bbfed2.

Seyiul committed:
fix: 비밀번호 유효성검사 수정

Seyiul committed:
Merge branch 'feat/PIM-42' into feat/PIM-44

Seyiul reviewed (approved):
와 멋져요 제 보드 화면에도 뽀려가야겠어요 👍 👍 👍 👍
